python/src/scipp/html/formatting_html.py

```diff
@@ -502,3 +502,27 @@ def human_readable_size(size_in_bytes):
         return f'{size_in_bytes/(1024):.2f} KB'
 
     return f'{size_in_bytes} Bytes'
+
+
+def inject_style():
+    """
+    Add our CSS style to the HTML head so that it can be used by all
+    HTML and SVG output without duplicating it in every cell.
+    This also preserves the style when the output in Jupyter is cleared.
+
+    The style is only injected once per session.
+    """
+    if not inject_style._has_been_injected:
+        from IPython.display import display, Javascript
+        from .resources import load_style
+        # `display` claims that its parameter should be a tuple, but
+        # that does not seem to work in the case of Javascript.
+        display(Javascript(f"""
```

Comment:
I'm not sure i understand why this has to be done by running some Javascript code, and not just using CSS classes.

Reply:
We need to put the CSS style sheet somewhere in the HTML document and it needs to be a parent of the stuff that we want to style. Thus we have to either repeat it in every output like we do before this PR. Or we place it in the HTML head or at least in some kind of notebook root element.
I don't know any way to do this in Python, Jupyter / IPython does not appear to expose any such functionality. But Javascript can do it just like you might do on a webpage.

Alternatively, we could link to a CSS file. But this is problematic because Jupyter limits file access to the directory you start it in and we cannot guarantee that the style sheet is visible form there.